stdlib/REPL/src/REPLCompletions.jl

```diff
@@ -481,7 +481,8 @@ CC.haskey(wvc::CC.WorldView{REPLInterpreterCache}, mi::MethodInstance) = haskey(
 CC.setindex!(wvc::CC.WorldView{REPLInterpreterCache}, ci::CodeInstance, mi::MethodInstance) = setindex!(wvc.cache.dict, ci, mi)
 
 # REPLInterpreter is only used for type analysis, so it should disable optimization entirely
-CC.may_optimize(::REPLInterpreter) = false
+# however, since effects are not computed until after optimization, the optimizations cannot be disabled
+CC.may_optimize(::REPLInterpreter) = true
```

Comment:
```suggestion
CC.may_optimize(::REPLInterpreter) = false
```
to fix the broken test case, because otherwise we may hit this case: https://github.com/JuliaLang/julia/blob/1abafe85b2c51417dc145c90e8dde72485293b6c/base/compiler/abstractinterpretation.jl#L790-L793

While it is true that post-opt analysis can refine effects, I doubt the benefits of that refinement really make up for the cost of going through all those optimization passes. 
Since post-opt analysis refinement doesn't really widen the effects, I think it's fine to cache the effects that come from abstract interpretation.